src/sagemaker_xgboost_container/algorithm_mode/serve.py

```diff
@@ -145,6 +154,51 @@ def execution_parameters():
     return flask.Response(response=response_text, status=http.client.OK, mimetype="application/json")
 
 
+def _parse_accept(request):
+    """Gets the accept type for a given request.
+
+    Valid accept types are "application/json", "application/jsonlines", "application/x-recordio-protobuf",
+    and "text/csv". If no accept type is set, use the value in SAGEMAKER_DEFAULT_INVOCATIONS_ACCEPT.
+
+    :param request: flask request
+    :return: parsed accept type
+    """
+    accept, _ = cgi.parse_header(request.headers.get("accept").lower())
+    if not any(accept in mimetypes for mimetypes in ["application/json", "application/jsonlines",
+                                                     "application/x-recordio-protobuf", "text/csv"]):
+        if sm_env_constants.SAGEMAKER_DEFAULT_INVOCATIONS_ACCEPT in os.environ:
+            return os.getenv('SAGEMAKER_DEFAULT_INVOCATIONS_ACCEPT')
+        raise ValueError("Accept type {} is not supported.".format(accept))
+    return accept
+
+
+def _handle_selectable_inference_response(predictions, accept):
+    """Retrieve the additional prediction data for selectable inference mode.
+
+    :param predictions: output of xgboost predict (list of numpy objects)
+    :param accept: requested accept type (str)
+    :return: flask response with encoded predictions
+    """
+    try:
+        # get objective function and content keys needed to make output
```

Comment:
These comments, while well intentioned, are unnecessary. The clearest code should not need inline documentation. Remove.